obspy/core/stream.py

```diff
@@ -3118,6 +3119,85 @@ def remove_sensitivity(self, *args, **kwargs):
             tr.remove_sensitivity(*args, **kwargs)
         return self
 
+    def stack(self, group='all', type='linear', npts_tol=0):
+        """
+        Return stream with traces stacked by the same selected metadata.
+
+        :param group: Stack waveforms together which have the same metadata
```

Comment:
I recommend also adding `:type group: str` so it is clear this must be a string. Same for the other parameters.